examples/linear_model/plot_logistic_multinomial.py

```diff
@@ -1,70 +1,183 @@
     else:
-        coef = clf.coef_
-        intercept = clf.intercept_
+        coef = classifier.coef_
+        intercept = classifier.intercept_
 
-    def plot_hyperplane(c, color):
-        def line(x0):
-            return (-(x0 * coef[c, 0]) - intercept[c]) / coef[c, 1]
+    for i in range(coef.shape[0]):
+        w = coef[i]
+        a = -w[0] / w[1]
+        xx = np.linspace(xmin, xmax)
+        yy = a * xx - (intercept[i]) / w[1]
+        ax.plot(xx, yy, "--", linewidth=3, label=f"Class {i}")
 
-        plt.plot([xmin, xmax], [line(xmin), line(xmax)], ls="--", color=color)
+    return ax.get_legend_handles_labels()
 
-    for i, color in zip(clf.classes_, colors):
-        plot_hyperplane(i, color)
+
+# %%
+fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(12, 5), sharex=True, sharey=True)
+
+for model, title, ax in [
+    (
+        logistic_regression_multinomial,
+        "Multinomial Logistic Regression Hyperplanes",
+        ax1,
+    ),
+    (logistic_regression_ovr, "One-vs-Rest Logistic Regression Hyperplanes", ax2),
+]:
+    hyperplane_handles, hyperplane_labels = plot_hyperplanes(model, X, ax)
+    scatter = ax.scatter(X[:, 0], X[:, 1], c=y, edgecolor="k")
+    scatter_handles, scatter_labels = scatter.legend_elements()
+
+    all_handles = hyperplane_handles + scatter_handles
+    all_labels = hyperplane_labels + scatter_labels
+
+    ax.legend(all_handles, all_labels, title="Classes")
+    ax.set_title(title)
 
 plt.show()
+
+# %%
+# While the hyperplanes for classes 0 and 2 are quite similar between the two methods,
+# we observe that the hyperplane for class 1 is notably different. This difference stems
+# from the fundamental approaches of one-vs-rest and multinomial logistic regression:
+#
+# For one-vs-rest logistic regression:
+#
+# - Each hyperplane is determined independently by considering one class against all
+#   others.
+# - For class 1, the hyperplane represents the decision boundary that best separates
+#   class 1 from the combined classes 0 and 2.
+# - This binary approach can lead to simpler decision boundaries but may not capture
+#   complex relationships between all classes simultaneously.
+#
+# For multinomial logistic regression:
+#
+# - All hyperplanes are determined simultaneously, considering the relationships between
+#   all classes at once.
+# - Each hyperplane represents the decision boundary where the probability of one class
+#   becomes higher than the others, based on the overall probability distribution.
+# - This approach can capture more nuanced relationships between classes, potentially
+#   leading to more accurate classification in multi-class problems.
+#
+# The difference in hyperplanes, especially for class 1, highlights how these methods
+# can produce different decision boundaries despite similar overall accuracy. The choice
+# between one-vs-rest and multinomial logistic regression can depend on the specific
```

Comment:
I disagree. We should generally recommend the multinomial version of LogReg.
In the rare case that ovr is better in some relevant metric, it is either random (bad) luck or a bad metric.
For predicting probabilities, multinomial is the crystal clear choice.

Therefore, I would appreciate a statement at the beginning that for educational purposes we compare with ovr.

Reply:
Agreed. I was not sure how to conclude the example here.

Now that you raise this point, I changed it to emphasize that the decision planes are just arbitrary position in the OvR while I assume that you can better craft a utility function that use the probabilities estimated from the multinomial LR and thus lead optimize the real problem that you have at hand.

If you think that we should not mention it (or that I overlooked something), I'm happy to only mentioned that you should use the multinomial case.